File: sdk/dotnet-sdk/src/VeritySDK.Test/Protocols/AbstractProtocolTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Json;
using VeritySDK.Protocols;
using VeritySDK.Transports;
using VeritySDK.Utils;

namespace VeritySDK.Test
{
    [TestClass]
    public class AbstractProtocolTest
    {
        [TestMethod]
        public void testGetMessage()
        {
            JsonObject message = JsonObject.Parse("{\"hello\": \"world\"}") as JsonObject;

            Context context = TestHelpers.getContext();
            byte[] packedMessage = AbstractProtocol.packMsg(context, message);

            JsonObject unpackedMessage = TestHelpers.unpackForwardMessage(context, packedMessage);

            Assert.AreEqual(unpackedMessage["hello"].ToString(), "\"world\"");

            TestHelpers.cleanup(context);
        }

    }
}

File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/protocols/AbstractProtocolTest.java
package com.evernym.verity.sdk.protocols;

import com.evernym.verity.sdk.TestHelpers;
import com.evernym.verity.sdk.utils.Context;
import org.json.JSONObject;
import org.junit.jupiter.api.Test;

import static com.evernym.verity.sdk.TestHelpers.unpackForwardMessage;
import static org.junit.jupiter.api.Assertions.assertEquals;

class SomeProtocol extends AbstractProtocol {
  private static final String MSG_QUALIFIER = "did:sov:some_did";
  private static final String MSG_FAMILY = "some-protocol";
  private static final String MSG_FAMILY_VERSION = "9.9.9";

  SomeProtocol() {
    super();
  }

  @Override
  public String qualifier() {
    return MSG_QUALIFIER;
  }

  @Override
  public String family() {
    return MSG_FAMILY;
  }

  @Override
  public String version() {
    return MSG_FAMILY_VERSION;
  }
}

public class AbstractProtocolTest {

  @Test
  public void testGetMessage() throws Exception {
    JSONObject message = new JSONObject("{\"hello\": \"world\"}");
    Context context = TestHelpers.getContext();
    byte[] packedMessage = AbstractProtocol.packMsg(context, message);
    JSONObject unpackedMessage = unpackForwardMessage(context, packedMessage);
    assertEquals(unpackedMessage.getString("hello"), "world");

    TestHelpers.cleanup(context);
  }
}